src/app/core/services/data.service.ts

```diff
@@ -208,17 +216,17 @@ export class DataService {
    * @param csvText - csv information.
    * @return - Subjects that tracks the parsed data.
    */
-  private csvParser(csvText: string): {}[] {
+  private csvParser(csvText: string): Array<ParsedUserData> {
```

Comment:
```suggestion
  private parseUserData(csvText: string): ParsedUserData[] {
```

We can consider using this representation instead. Also, we can change the method signature such that the method name accurately describes the purpose of this method. 

What are your thoughts on this? @zatkiller

Reply:
Sure, we can change the method name to indicate return type for greater clarity